### DISCUSSION TITLE
openssl 3.0.13 test failure 03-test_internal_modes, 30-test_evp, 30-test_evp_extra

### DISCUSSION THREAD

**Discussion by marchharesoftware** [openssl/openssl#24485] [Q&A]
HI!

Im building OpenSSL 3.0.13 on Windows 10 with 

perl Configure VC-WIN32 --prefix=D:/mydeps/openSSL  --openssldir=D:/mydeps/openSSL/ssl enable-crypto-mdebug enable-ssl3 enable-ssl3-method enable-tls1_3 no-ec

And I'm getting the following test failures:

Test Summary Report
-------------------
03-test_internal_modes.t         (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
30-test_evp.t                    (Wstat: 512 Tests: 64 Failed: 2)
  Failed tests:  2, 29
  Non-zero exit status: 2
30-test_evp_extra.t              (Wstat: 512 Tests: 3 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
Files=253, Tests=1870, 2755 wallclock secs (16.91 usr +  3.56 sys = 20.47 CPU)

[make_test_results_w32_openssl_3_0_13_try_2.txt](https://github.com/openssl/openssl/files/15425038/make_test_results_w32_openssl_3_0_13_try_2.txt)

I'm using an older toolchain, which could be the cause, but I'm wondering if anyone has any specific feedback on what may be the root cause these errors and/or if it's a genuine failure of the openssl build or a bug in the test?

build chain is:
Perl 5.22
nasm 2.12
Platform SDK 2003 R2
Microsoft 32-bit C/C++ Version 13.10.6030 

Thanks for any assistance!

**Comment by t8m** [comment]
There is apparently something broken with the GCM tag computation in your build. It could be a miscompilation by the old toolchain. It requires deeper investigation. I'd recommend updating the toolchain to something more recent at least to test whether the failures go away.

**Comment by marchharesoftware** [comment]
I've been trying a few combinations of things.

I tried upgrading the toolchain and checking I could build OK with a 'supported' chain, and that all worked OK (as expected).

I then tried with newer Perl and NASM but older x32 compiler - still failed, but then tried x64 (VC-WIN64A) and had troubles with the declspec(deprecated) syntax so ended up building with -DPEDANTIC -DOPENSSL_SUPPRESS_DEPRECATED  and that passed all OK.

So I ended up getting this working with x32 with:

perl Configure VC-WIN32 -DPEDANTIC -DOPENSSL_SUPPRESS_DEPRECATED -D_WIN32_WINNT=0x0502 --prefix=D:/cvsdeps/openSSL  --openssldir=D:/cvsdeps/openSSL/ssl enable-ssl3 enable-ssl3-method enable-tls1_3

I'm not sure if there is an interaction with declspec(deprecated) or it is the PEDANTIC define that solved it in the end, but it passes all tests now, so for that I'm happy to move on...

Hope this helps someone else in future, though it's a pretty narrow niche case.
